src/middlewared/middlewared/plugins/ups.py

```diff
@@ -191,10 +191,31 @@ def driver_choices(self):
         old_config['toemail'] = ';'.join(old_config['toemail']) if old_config['toemail'] else ''
 
         if len(set(old_config.items()) ^ set(config.items())) > 0:
+            if config['identifier'] != old_config['identifier']:
```

Comment:
But you are going to restart UPS service anyway and they will be dismissed there.

Reply:
I am not so sure about that - if the identifier changes - new identifier will be used to dismiss them and that wouldn't work ?